opensearch_py_ml/ml_models/metrics_correlation/dip.py

```diff
@@ -0,0 +1,407 @@
+# SPDX-License-Identifier: Apache-2.0
+# The OpenSearch Contributors require contributions made to
+# this file be licensed under the Apache-2.0 license or a
+# compatible open source license.
+# Any modifications Copyright OpenSearch Contributors. See
+# GitHub history for details.
+
+from typing import Tuple
+
+import torch
+
+
+def dip(x: torch.Tensor) -> float:
+    """
+    Fast computation of the dip test statistic. See: https://www.jstor.org/stable/pdf/2241144.pdf (paper)
```

Comment:
Both functions here need to be documented a little better. I can help with that.

Reply:
Yeah, I left few functions, parameters of the functions without comments where I'm hoping your help. The target of this PR is just to setup the skeleton so that we can start gradually fleshing it out.